tests/test_backend/test_mlir/test_mlir_compiler.py

```diff
@@ -0,0 +1,17 @@
+import magma as m
+
+
+def test_mlir_compiler_finalize_when_order():
```

Comment:
let's move this to an example in `examples.py` as the other ones, maybe named `complex_when_lazy_tuple`

Reply:
I moved it to examples, but it looks like the existing examples flow doesn't use the mlir_compiler interface, so it doesn't test that pass ordering, so I left it in this separate test that uses `m.compile` for now